### ISSUE TITLE
Service bind data does not contain service group members within the same supervisor

### ISSUE THREAD

**Issue by reset** [habitat-sh/habitat#2150]
Starting multiple services within the same supervisor with one or more services binding to a service group where one of the services within the same supervisor is a member will result in the templating engine not receiving that member. We need to debug this and make sure that services running within the same supervisor can bind to each other successfully.

**Comment by mwrock** [comment]
I think I ran into this yesterday addressing #1816. The test plans I were using had a:
```
{{#if bind.has_backend }}
{{~#each bind.backend.members}}
{{~#if alive }}
    server {{sys.ip}} {{sys.ip}}:{{cfg.port}}
{{~/if}}
{{~/each}}
{{~/if}}
```
and it was not binding. If I changed this to:

```
server {{bind.backend.members[0].sys.ip}} {{bind.backend.members[0].sys.ip}}:{{bind.backend.members[0].cfg.port}}
```
Things bound fine. So I think this is related to either `bind.has_`, the alive check or member iteration. I'm gonna dig more today.

**Comment by reset** [comment]
Closed by #2226
